### PR TITLE
Use new search filter in project autocomplete - PMT #99496

### PR DESCRIPTION
**Pull request by nikolas**
The most important change here is that we're now filtering
the projects with a `startswith` search instead of `contains`.
This gives better results, closer to what I think people expect.

I had to override a method in DRF's SearchFilter in order to
allow us to search on terms with spaces, (e.g. try searching
for "project m" to find the "Project Management Tool" project).
I might make a pull request upstream if I find no existing way to
configure this.

### CODE REVIEW

**Comment by thraxil** [comment]
If it searches *only* from the beginning, than I'm going to vote against it. A lot of our projects have long names and then (ACRONYM) at the end and ACRONYM is what most of us actually know the project by or are otherwise not necessarily named in ways that startswith is very helpful for. Some sample use-cases:

* searching for "CARE" needs to still turn up "Child Abuse Reporting Education (CARE)"
* searching for "UDL" still needs to turn up "Chemistry Instrument Lab Environment (UDL)"
* searching for "forest" still needs to turn up "Django Forest (Directed Learning CMS)"

Basically, I think search still needs to be via "contains", but ideally, search results would be ordered with startswith matches first.

**Comment by nikolas** [comment]
Hmm, yeah good point. I will figure out a way to order the results to have `startswith` results at the beginning. My initial attempts where in typeahead.js, (really bloodhound), but I think actually this should be done on the back-end.

**Comment by nikolas** [comment]
@thraxil I've revised this PR - it now sorts the search results correctly by putting the results that start with the term at the beginning.

The `contains` search is still present, so searching for `CARE`, `UDL`, and `forest` all work as expected.
